docs/concepts/Dialog-Command.md

```diff
@@ -0,0 +1,78 @@
+# Dialog Command
+
+`askx dialog` opens a interactive mode, similar to Node's REPL, in which developers can simulate a multi-turn conversation with  Alexa. For each utterance, Dialog command leverages [simulate-skill](https://developer.amazon.com/en-US/docs/alexa/smapi/ask-cli-command-reference.html#simulate-command) to asynchronously poll simulation result until it's available and shows Alexa text responses. **Note:** Skill must have been enabled for simulation before running the dialog command.
+
+Structure of the dialog command is:
+
+`askx dialog [-s|--skill-id <skill-id>] [-l|--locale <locale>] [-g|--stage <stage>] [-r|--replay <file-path>] [-p|--profile <profile>] [--debug]`
+
+Default value for `--profile`is `default`and `--stage` is `development`. Specifying `--debug` appends a debug message to the standard error.
+
+##Dialog command supports two modes:
+
+### INTERACTIVE MODE
+
+This mode allows developers to simulate a multi-turn conversation with Alexa. If `skill-id` is not specified, the command must be run from the skill project root directory. If the environment variable with key `ASK_DEFAULT_DEVICE_LOCALE` is not specified, then locale must be provided as part of arguments. Alternatively, User has an option to record the list of utterances so far in a JSON file, by using a special command `.record`, which will be used in replay mode. User may either keep interacting with the skill or exit the Interactive mode by using a special command, `.quit`.
+
+User experience in Interactive mode will be,
+
+```
+============================== Welcome to ASK Dialog ===============================
+=== In interactive mode, type your utterance text onto the console and hit enter ===
+============= Alexa will then evaluate your input and give a response! =============
+
+User  > open hello world
+Alexa > Welcome, you can say Hello or Help. Which would you like to try?
+User  > help
+Alexa > You can say hello to me! How can I help?
+User  > hello
+Alexa > Hello World!
+User  > .record <path_to_replay_file>
+Created replay file at <path_to_replay_file>
+User  > .quit
+
+================================= Goodbye! =========================================
+```
+#### Sample replay file format:
```

Comment:
I suggest let this bullet point be the introduction for the reply file, and put it into the Replay Mode section as it serves the replay mode mostly. 
And how about changing this title to "~~Sample~~ Format for the Replay file"? The reason is we need to briefly describe what each field does in this file, like the `userInput` we actually support special command. 
You can still say the JSON example is a sample file from the utterance above.

Reply:
I'm not so sure about this. 
We create the Replay file in interactive mode. Hence the depiction here.
User can relate to *what was the interaction made in Interactive Mode* and compare it with what was recorded in Replay file immediately. Moving it to Replay Mode might throw off the reader as to what happens when he/she runs `.record` sub-sommand.